cmd/dagger/shell_commands.go

```diff
@@ -439,6 +440,41 @@ func (h *shellCallHandler) registerCommands() { //nolint:gocyclo
 Writes any specified operands, separated by single blank (' ') characters and followed by a newline ('\n') character, to the standard output. If the -n option is specified, the trailing newline is suppressed.
 `,
 		},
+		&ShellCommand{
+			Use:  ".env [name]",
+			Args: MaximumArgs(1),
+			Description: `Show available environment variables or a specific variable
+
+If no name is provided, all environment variables are printed. If a name is provided, the value of that environment variable is printed.
+			`,
+			State: NoState,
+			Run: func(ctx context.Context, cmd *ShellCommand, args []string, _ *ShellState) error {
+				if len(args) == 0 {
+					// Print all environment variables
+					env := os.Environ()
+					if len(env) == 0 {
+						return h.Print(ctx, "No environment variables set")
+					}
+
+					return h.Print(ctx, strings.Join(env, "\n"))
+				}
```

Comment:
Instead of `os.Environ()` I think it's better to use the interpreter's own env vars. It'll reflect more precisely what vars are available in the script, even user defined ones. The host env vars are just the default when the shell starts. 
```suggestion
				hc := interp.HandlerCtx(ctx)
				if len(args) == 0 {
					// Print all environment variables
					hc.Env.Each(func(name string, v expand.Variable) bool {
						fmt.Fprintf(hc.Stdout, "%s=%s\n", name, v.String())
						return true
					})
					return nil
				}

```